### PR TITLE
[HGCal] Modify MultiTrackValidator plotting for easier browsing of HGCal plots

### PR DESCRIPTION
**Pull request by apsallid**
#### PR description:

For the plotting part of the HGCalValidator package we take advantage of the tools already created by the MultiTrackValidator. 
However, one option was added that when set to true it will produce a webpage with all plots to be able to 
visualize all images at a glance. 

#### PR validation:

This PR depends on PR #26099. So, after the harvesting step, you can run the makeHGCalValidationPlots.py script with the new separate option e.g: 

```
python Validation/HGCalValidation/python/makeHGCalValidationPlots.py DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root --outputDir HGCValid --no-ratio --png --separate --html-sample Samplename --html-validation-name hgcalLayerClusters --subdirprefix test1
```

@felicepantaleo @rovere @cseez @malgeri 

@makortel Could you please comment?

### CODE REVIEW

File: Validation/RecoTrack/python/plotting/plotting.py [hunk 1]
```diff
@@ -1964,7 +1964,7 @@ def _doStats(h, col, dy):
             st.SetX1NDC(startingX)
             st.SetX2NDC(startingX+0.3)
             st.SetY1NDC(startingY+dy)
-            st.SetY2NDC(startingY+dy+0.15)
+            st.SetY2NDC(startingY+dy+0.25)
```
Comment (makortel) [inline Validation/RecoTrack/python/plotting/plotting.py]: This change increases the height of the stat box, right? IIRC the space between stat boxes of different histograms/graphs was already rather small. Can you show that this change does not lead the stat boxes to overlap?

File: Validation/RecoTrack/python/plotting/plotting.py [hunk 2]
```diff
@@ -2532,6 +2532,7 @@ def create(self, tdirectoryNEvents, requireAllHistograms=False):
     def draw(self, *args, **kwargs):
         kargs = copy.copy(kwargs)
         kargs["ratio"] = False
+        kargs["separate"] = False
```
Comment (makortel) [inline Validation/RecoTrack/python/plotting/plotting.py]: Could you explain why this change is needed?

File: Validation/RecoTrack/python/plotting/validation.py [hunk 3]
```diff
@@ -1293,5 +1293,14 @@ def _doPlots(self, plotterFolder, dqmSubFolder):
             print("Typically this is a naming problem in the plotter configuration")
             sys.exit(1)
 
+        if (self._plotterDrawArgs.get("separate", False) == True):
```
Comment (makortel) [inline Validation/RecoTrack/python/plotting/validation.py]: I'd rather go with
```suggestion
        if self._plotterDrawArgs.get("separate", False):
```

File: Validation/RecoTrack/python/plotting/validation.py [hunk 4]
```diff
@@ -1293,5 +1293,14 @@ def _doPlots(self, plotterFolder, dqmSubFolder):
...
+            if not os.path.exists("%s/res"%newdir):
+              os.makedirs("%s/res"%newdir)
+            downloadables = ["index.php", "res/jquery-ui.js", "res/jquery.js", "res/style.css", "res/style.js", "res/theme.css"]
+            for d in downloadables:
+                if not os.path.exists("%s/%s" % (newdir,d)):
+                    import urllib
```
Comment (makortel) [inline Validation/RecoTrack/python/plotting/validation.py]: Could the import be moved to the top of the file?

**Review by makortel — COMMENTED** [review]
Could you show an example of how the output directory looks like (in the web)?

**Comment by apsallid** [comment]
Thanks a lot for the super fast reply! This is how the output looks like in the web: 

http://apsallid.web.cern.ch/apsallid/HGCalValidationPlots/

And if you click on any of these, you will see at the end a Browse Folder button.

File: Validation/RecoTrack/python/plotting/plotting.py (hunk 2 above)
Comment (rovere) [reply Validation/RecoTrack/python/plotting/plotting.py]: Ciao Matti,
the separate option was giving conflicts (crashes, IIRC) if coupled with the PlotOnSideGroup, which sort of makes sense since the 2 concepts are somehow "opposite". That's why I defaulted that option to False in this specific case. I guess you never tested the 2 coupled together, right?

Comment (makortel) [reply Validation/RecoTrack/python/plotting/plotting.py]: It's indeed very likely that I did not test these two together, and on a further thought I do agree that the "separate" concept doesn't really apply to the PlotOnSideGroup. Thanks for the explanation!

File: Validation/RecoTrack/python/plotting/plotting.py (hunk 1 above)
Comment (makortel) [reply Validation/RecoTrack/python/plotting/plotting.py]: From your example I see that your case has indeed more lines in the stat box than MTV has (by default). I'm not against that, but then the height should be made e.g. configurable or automated, and the height should propagate to the "line height" (magical `dy -= 0.19` below) as well.

**Comment by makortel** [comment]
Looks good to me, thanks.

**Comment by makortel** [comment]
@cmsbuild, please test

(doesn't really test anything but anyway)

**Comment by fabiocos** [comment]
please test

**Comment by alberto-sanchez** [comment]
+1

**Comment by andrius-k** [comment]
+1

**Comment by fabiocos** [comment]
+1
